File: DemoServer/AdditionalLanguages/go/TextSearch/FTSWithStaticIndexMultipleFields/FTSWithStaticIndexMultipleFields.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    createDatabase()
    fullTextSearchMultipleField("Floyd")
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

func createDatabase() {
    databaseRecord := ravendb.NewDatabaseRecord()
    databaseRecord.DatabaseName = "testGO"
    createDatabaseOperation := ravendb.NewCreateDatabaseOperation(databaseRecord, 1)
    var err = globalDocumentStore.Maintenance().Server().Send(createDatabaseOperation)
    if err != nil {
        fmt.Printf("d.store.Maintenance().Server().Send(createDatabaseOperation) failed with %s\n", err)
    }
}

//region Demo
func fullTextSearchMultipleField(searchTerm string) error {

    //region Step_1
    indexName := "Song/TextData"
    index := ravendb.NewIndexCreationTask(indexName)
    //endregion
 
    //region Step_2
    index.Map = `
        from song in docs.LastFms
        select new {
             SongData = new {
                song.Artist,
                song.Title,
                song.Tags,
                song.TrackId
             }
        }`
    //endregion
    
    //region Step_3    
    index.Index("SongData", ravendb.FieldIndexingSearch)
    //endregion
 
    //region Step_4
    err := index.Execute(globalDocumentStore, nil, "")
    if err != nil {
        return err
    }
    //endregion
 
    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()
 
    //region Step_5
    query := session.QueryIndex(indexName)
    query = query.Search("SongData", searchTerm)
    query = query.Take(20)
 
    var results []*LastFm
    err = query.GetResults(&results)
    if err != nil {
        return err
    }
    //endregion
    
    return nil
 }

type LastFm struct {
    ID        string
    Artist    string
    Title     string
    TrackID   string
    Tags      []string
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSWithStaticIndexMultipleFields/FTSWithStaticIndexMultipleFields.java
package net.ravendb.demo.textSearch.fTSWithStaticIndexMultipleFields;

//region Usings
import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;
import net.ravendb.client.documents.indexes.FieldIndexing;
import net.ravendb.client.documents.session.IDocumentSession;
import java.util.List;
//endregion
import net.ravendb.demo.common.DocumentStoreHolder;
import net.ravendb.demo.common.models.LastFm;

public class FTSWithStaticIndexMultipleFields {

    //region Demo
    //region Step_1
    public static class Song_TextData extends AbstractIndexCreationTask {
    //endregion

        public Song_TextData() {
            //region Step_2
            map = "docs.LastFms.Select(song => new { " +
                "    songData = new object[] { " +
                "        song.Artist, " +
                "        song.Title, " +
                "        song.Tags, " +
                "        song.TrackId " +
                "    } " +
                "})";
            //endregion

            //region Step_3
            index("songData", FieldIndexing.SEARCH);
            //endregion
        }
    }
    //endregion

    public List<LastFm> run(RunParams runParams) {
        String searchTerm = runParams.getSearchTerm();

        //region Demo
        List<LastFm> results;

        try (IDocumentSession session = DocumentStoreHolder.mediaStore.openSession()) {
            //region Step_4
            results = session.query(LastFm.class, Song_TextData.class)
                .search("songData", searchTerm)
                .take(20)
                .toList();
            //endregion
        }
        //endregion

        return results;
    }

    public static class RunParams {
        private String searchTerm;

        public String getSearchTerm() {
            return searchTerm;
        }

        public void setSearchTerm(String searchTerm) {
            this.searchTerm = searchTerm;
        }
    }
}
